checkov/secrets/plugins/detector_utils.py

```diff
@@ -0,0 +1,252 @@
+        key=KEY,
+        whitespace=OPTIONAL_WHITESPACE,
+        quote=QUOTE,
+        secret=SECRET,
+    ),
+    flags=re.IGNORECASE,
+)
+
+#  if the current regex is not enough, can add more regexes to check
+
+YML_PAIR_VALUE_KEYWORD_REGEX_TO_GROUP = {
+    FOLLOWED_BY_COLON_VALUE_KEYWORD_REGEX: 4,
+}
+
+YML_PAIR_VALUE_SECRET_REGEX_TO_GROUP = {
+    FOLLOWED_BY_COLON_VALUE_SECRET_REGEX: 4,
+}
+
+JSON_PAIR_VALUE_KEYWORD_REGEX_TO_GROUP = {
+    QUOTES_REQUIRED_FOLLOWED_BY_COLON_VALUE_KEYWORD_REGEX: 4,
+}
+
+JSON_PAIR_VALUE_SECRET_REGEX_TO_GROUP = {
+    QUOTES_REQUIRED_FOLLOWED_BY_COLON_VALUE_SECRET_REGEX: 4,
+}
+
+TERRAFORM_PAIR_VALUE_KEYWORD_REGEX_TO_GROUP = {
+    FOLLOWED_BY_EQUAL_VALUE_KEYWORD_REGEX: 4,
+}
+
+TERRAFORM_PAIR_VALUE_SECRET_REGEX_TO_GROUP = {
+    FOLLOWED_BY_EQUAL_VALUE_SECRET_REGEX: 4,
+}
+
+REGEX_VALUE_KEYWORD_BY_FILETYPE = {
+    FileType.YAML: YML_PAIR_VALUE_KEYWORD_REGEX_TO_GROUP,
+    FileType.JSON: JSON_PAIR_VALUE_KEYWORD_REGEX_TO_GROUP,
+    FileType.TERRAFORM: TERRAFORM_PAIR_VALUE_KEYWORD_REGEX_TO_GROUP,
+}
+
+REGEX_VALUE_SECRET_BY_FILETYPE = {
+    FileType.YAML: YML_PAIR_VALUE_SECRET_REGEX_TO_GROUP,
+    FileType.JSON: JSON_PAIR_VALUE_SECRET_REGEX_TO_GROUP,
+    FileType.TERRAFORM: TERRAFORM_PAIR_VALUE_SECRET_REGEX_TO_GROUP,
+}
+
+SINGLE_LINE_PARSER = {
+    FileType.TERRAFORM: terraform_single_line_parser,
+}
+
+MULTILINE_PARSERS = {
+    FileType.YAML: (
+        (FileType.YAML, yml_multiline_parser),
+    ),
+    FileType.JSON: (
+        (FileType.JSON, json_multiline_parser),
+    ),
+    FileType.TERRAFORM: (
+        (FileType.TERRAFORM, terraform_multiline_parser),
+        (FileType.JSON, json_multiline_parser),
+        (FileType.YAML, yml_multiline_parser),
+    ),
+}
+
+
+def remove_fp_secrets_in_keys(detected_secrets: set[PotentialSecret], line: str) -> None:
+    for detected_secret in detected_secrets:
+        if detected_secret.secret_value and line.replace('"', '').replace("'", '').startswith(
+                detected_secret.secret_value):
+            # Found keyword prefix as potential secret
+            detected_secrets.remove(detected_secret)
+            break
+
+
+def format_reducing_noise_secret(string: str) -> str:
+    return json.dumps(string)
+
+
+def extract_from_string(pattern: dict[Pattern[str], int] | None, string: str) -> set[str]:
+    matches: set[str] = set()
+    if not pattern:
+        return matches
+    for value_regex, group_number in pattern.items():
+        match = value_regex.search(string)
+        if match:
+            matches |= {match.group(group_number)}
+    return matches
+
+
+def detect_secret(
+        scanners: tuple[HighEntropyStringsPlugin, ...],
+        filename: str,
+        line: str,
+        line_number: int = 0,
+        **kwargs: Any,
+) -> set[PotentialSecret]:
+    for entropy_scanner in scanners:
+        matches = entropy_scanner.analyze_line(filename, line, line_number, **kwargs)
```

Comment:
if this method is going to be static, it might use other scanners then entropy-based, so it should not use the naming it had previously when it was part of the `EntropyKeywordCombinator` class.

Reply:
Can you clarify this? The method isn't static, and I want to leave naming as-is to prevent breaking the consistency of using the same method name for methods with similar purpose (we use it in other parts of the code as well, not just entropy)